lib/matplotlib/tri/tripcolor.py

```diff
@@ -105,15 +131,17 @@ def tripcolor(ax, *args, alpha=1.0, norm=None, cmap=None, vmin=None,
         # Color values.
         if facecolors is None:
             # One color per triangle, the mean of the 3 vertex color values.
-            C = C[maskedTris].mean(axis=1)
+            colors = point_colors[maskedTris].mean(axis=1)
         elif tri.mask is not None:
             # Remove color values of masked triangles.
-            C = C[~tri.mask]
+            colors = facecolors[~tri.mask]
+        else:
+            colors = facecolors
 
         collection = PolyCollection(verts, **kwargs)
 
     collection.set_alpha(alpha)
-    collection.set_array(C)
+    collection.set_array(colors)
```

Comment:
While you're at it, probably kwargs should be set last, for the same reason as in #21146/#21525?

Reply:
But there is `collection._scale_norm(norm, vmin, vmax)` after setting the norm here. AFAICS this handles the issue in #21146/#21525. One can probably remove that when delaying kwargs setting. But since this PR is already complicated enough and there is no immediate problem, I'd not go into this within this PR.